luigi/worker.py

```diff
@@ -159,13 +159,15 @@ def run(self):
             self.task.trigger_event(Event.FAILURE, self.task, ex)
             subject = "Luigi: %s FAILED" % self.task
 
-            error_message = notifications.wrap_traceback(self.task.on_failure(ex))
+            raw_error_message = self.task.on_failure(ex)
+            notification_error_message = notifications.wrap_traceback(raw_error_message)
+            expl = json.dumps(raw_error_message)
```

Comment:
You're right, I forgot to mention this. I think this is the right behavior because we do it for `on_success` (i.e. `json.dumps(self.task.on_success())`).